src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/utils/rule/validation/SimpleSwitchRuleConverter.java

```diff
@@ -114,9 +119,26 @@ private SimpleSwitchRule buildIngressSimpleSwitchRule(Flow flow, FlowPath flowPa
             } else if (flow.getEncapsulationType().equals(FlowEncapsulationType.VXLAN)) {
                 rule.setTunnelId(encapsulationId.getEncapsulationId());
             }
+
+            if (ingress.isLooped() && !flowPath.isProtected()) {
+                rules.add(buildIngressLoopSimpleSwitchRule(rule, flowPath, ingress));
```

Comment:
We are making the same rule for single-switch and for multi-switch flows is it expected behavior?

Reply:
Ingress loop rule is dependent only on input port and vlan stack so it's the same for single-switch and multi-switch flows.